go.mod

```diff
@@ -221,8 +233,13 @@ replace (
 	// Use special SDK v0.47.x release with support for both ICS and LSM
 	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.47.10-ics-lsm
 
+	// use memiavl for faster storage
+	github.com/cosmos/cosmos-sdk/store => github.com/crypto-org-chain/cronos/store v0.0.5-0.20240301012710-e9bb006c8d14
+	github.com/cosmos/cosmos-sdk/versiondb => github.com/crypto-org-chain/cronos/versiondb v0.0.5-0.20240301012710-e9bb006c8d14
```

Comment:
this one is not needed for memiavl alone.

Reply:
> Would it be useful for RPC/API providers that need to be able to provide query results with data from different heights?

It do support historical state/proof queries, and the supported range is determined by the `snapshot-keep-recent` config, it support historical query from the oldest available snapshot to latest height.

Reply:
But I have to say, historical query is not that efficient, I'm not sure how it performs under load, we use versiondb aka. state storage to provide state queries, only use memiavl to provide proof queries.

Reply:
If historical state/proof query support is removed (no go mod replace), will it case any problems for validators or users of API/RPC?

Reply:
Ah, you mean versiondb, yes, it's dedicated to supporting efficient historical state queries. it's a different name for state storage of store v2 design.

Reply:
Lets keep versiondb to help the RPC node providers.  The store mod replace will be removed.

Reply:
> Or is this only useful for creating statesync snapshots on a verison that memiavl does not snapshot?

You can't create state sync snapshot from versiondb, because it don't have information for the branch nodes, only leave key value pairs.
and memiavl do support creating state sync snapshots for all the intermediate versions between the oldest snapshot and latest height.

Reply:
Thank you for the clarification

Reply:
> Thank you for the clarification. For others reviewing, we discussed how to enable snapshots for intermediate versions here: [cosmos/iavl#878 (comment)](https://github.com/cosmos/iavl/pull/878#issuecomment-1970310657)

maybe you can add some comments for the individual parameters here for now.

Reply:
Comments on parameters added